### PR TITLE
[WAF] Add resource huaweicloud_waf_dedicated_domain to manage WAF dedicated domains.

### LINKED ISSUE #1355: [WAF] Add data sources and resources for dedicated engine.
### Current Terraform Version

```
1.27.0
```

This is an internal requirement as follows:
- Add the data source to query WAF policies, source name is `huaweicloud_waf_policies`.
- Manage domain names in Premium-waf, resource name is `huaweicloud_waf_dedicated_domain`.
  Functions include: creating a domain names, modifying protection status, modifying protection mode, and deleting domain names.
- Manage engine in Premium-waf, resource name is `huaweicloud_waf_dedicated_engine`.
  Functions include: creating a dedicated engine,deleting dedicated,querying dedicated engine by id.
- We also needs to query the dedicated engine infomation, the data source name is `huaweicloud_waf_dedicated_engines` and `huaweicloud_waf_dedicated_engine`.

### PR DESCRIPTION
**Pull request by chengxiangdong**
**What this PR does / why we need it**:

- Add the resource to manage WAF dedicated engines, resource name is `huaweicloud_waf_dedicated_domain`.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1355

**Special notes for your reviewer**:

* The endpoint and `WafDedicatedV1Client`  are shared with waf dedicated instance, padding for https://github.com/huaweicloud/terraform-provider-huaweicloud/pull/1407 to be merged.

**Release note**:

```release-note
NONE
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccWafDedicateDomainV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccWafDedicateDomainV1_basic -timeout 360m -parallel 4
=== RUN   TestAccWafDedicateDomainV1_basic
=== PAUSE TestAccWafDedicateDomainV1_basic
=== CONT  TestAccWafDedicateDomainV1_basic
--- PASS: TestAccWafDedicateDomainV1_basic (489.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       489.445s

```

### CODE REVIEW

File: huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_domain.go [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+}
+
+// getCertificateNameById get certificate id from HuaweiCloud.
+func getCertificateNameById(d *schema.ResourceData, meta interface{}) (string, error) {
+	config := meta.(*config.Config)
+	c, err := config.WafV1Client(config.GetRegion(d))
+	if err != nil {
+		return "", fmtp.Errorf("error creating HuaweiCloud WAF Client: %s", err)
+	}
+	if len(d.Get("certificate_id").(string)) > 0 {
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_domain.go]: use d.GetOk instead

File: huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_domain.go [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+		d.Set("cipher", dm.Cipher),
+	)
+	if mErr.ErrorOrNil() != nil {
+		return fmtp.Errorf("error setting WAF fields: %s", err)
+	}
+
+	// The resources of compliance_certification, alarm_page and traffic_identifier may be empty.
+	d.Set("compliance_certification", buildComplianceCertificationAttribute(dm))
+	d.Set("traffic_identifier", buildTrafficIdentifierAttribute(dm))
+	d.Set("alarm_page", buildAlarmPageAttribute(dm))
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_domain.go:324]: these lines can also append in multierror

File: docs/resources/waf_dedicated_domain.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+* `certificate_id` - (Optional, String) Specifies the certificate ID. This parameter is mandatory when `client_protocol` 
+  is set to HTTPS.
+  
+* `policy_id` - (Optional, String) Specifies the policy ID associated with the domain.
+  If not specified, a new policy will be created automatically. 
+
+* `proxy` - (Optional, Bool) Specifies whether a proxy is configured. Default value is `false`.
+
+  -> **Note:**
```
Comment (ShiChangkuo) [inline docs/resources/waf_dedicated_domain.md]: format error, please preview the docs by https://registry.terraform.io/tools/doc-preview

**Review by ShiChangkuo — CHANGES_REQUESTED** [review]

File: huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_domain.go (hunk 1 above)
Comment (chengxiangdong) [reply huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_domain.go]: ok

File: docs/resources/waf_dedicated_domain.md (hunk 3 above)
Comment (chengxiangdong) [reply docs/resources/waf_dedicated_domain.md]: fixed

File: huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_domain.go (hunk 2 above)
Comment (chengxiangdong) [reply huaweicloud/services/waf/resource_huaweicloud_waf_dedicated_domain.go:324]: usually the values of `compliance_certification`, `traffic_identifier` and `alarm_page` are empty.

**Review by ShiChangkuo — APPROVED** [review]

**Pull request merged**